workloads/upgrade-perf/rosa_upgrade.sh

```diff
@@ -0,0 +1,138 @@
+    echo "ERROR: Cluster ${ROSA_CLUSTER_NAME} not found on rosa list clusters results. Exiting..."
+    exit 1
+  fi
+  if [ ! -z ${LATEST} ] ; then
+    VERSION=$(rosa list upgrade -c ${ROSA_CLUSTER_NAME} | sed -e 1d | head -1 | awk '{print $1}')
+  elif [ ! -z ${TOVERSION} ] ; then
+    VERSION=$(rosa list upgrade -c ${ROSA_CLUSTER_NAME} | sed -e 1d | awk '{print $1}' | grep ${TOVERSION})
+  else
+    echo "ERROR: Missing upgrade strategy. No LATEST or TOVERSION are defined. Exiting..."
+    exit 1
+  fi
+  if [ -z ${VERSION} ] ; then
+    echo "ERROR: No version to upgrade found running rosa list upgrade -c ${ROSA_CLUSTER_NAME}"
+    exit 1
+  fi
+  CURRENT_VERSION=$(oc get clusterversion | grep ^version | awk '{print $2}')
+  rosa upgrade cluster -c ${ROSA_CLUSTER_NAME} --version=${VERSION} -y --schedule-date $(date +%Y-%m-%d) --schedule-time $(date -u --date="+6 minutes" +%H:%M)
+  # Sleep 6 minutes, rosa upgrade dont let to schedule an upgrade in less than 5 minutes from now
+  sleep 360
+  oc delete pods -n openshift-insights --all
+  oc delete pods -n openshift-managed-upgrade-operator --all
+  rosa_control_plane_upgrade_active_waiting ${VERSION}
+  if [ $? -eq 0 ] ; then
+    CONTROLPLANE_UPGRADE_RESULT="OK"
+  else
+    CONTROLPLANE_UPGRADE_RESULT="Failed"
+  fi
+  rosa_workers_active_waiting
+  if [ $? -eq 0 ] ; then
+    WORKERS_UPGRADE_RESULT="OK"
+  else
+    WORKERS_UPGRADE_RESULT="Failed"
+  fi
+  rosa_upgrade_index_results ${CONTROLPLANE_UPGRADE_DURATION} ${CONTROLPLANE_UPGRADE_RESULT} ${WORKERS_UPGRADE_DURATION} ${WORKERS_UPGRADE_RESULT} ${CURRENT_VERSION} ${VERSION}
+  exit 0
+}
+
+rosa_workers_active_waiting(){
+  start_time=$(date +%s)
+  WORKERS=$(oc get node --no-headers -l node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",node-role.kubernetes.io/worker="" 2>/dev/null | wc -l)
+  # Giving 5 minutes per worker
+  ITERATIONWORKERS=0
+  while [ ${ITERATIONWORKERS} -le $(( ${WORKERS}*5 )) ] ; do
+    if [ $(rosa describe cluster -c ${ROSA_CLUSTER_NAME} | grep ^Scheduled | wc -l) -ne 1 ] ; then
+      echo "INFO: Upgrade finished for ROSA, continuing..."
+      end_time=$(date +%s)
+      export WORKERS_UPGRADE_DURATION=$((${end_time} - ${start_time}))
+      return 0
+    else
+      LASTSTATUS=$(oc logs $(oc get pods -n openshift-managed-upgrade-operator -o Name | grep -v catalog) -n openshift-managed-upgrade-operator | grep "workers are upgraded" | tail -1)
+      echo "INFO: ${ITERATIONWORKERS}/$(( ${WORKERS}*5 )). Last Update: ${LASTSTATUS}."
+      echo "INFO: Waiting 60 seconds for the next check..."
+      ITERATIONWORKERS=$((${ITERATIONWORKERS}+1))
+      sleep 60
+    fi
+  done
+  echo "ERROR: ${ITERATIONWORKERS}/$(( ${WORKERS}*5 )). ROSA workers upgrade not finished after $(( ${WORKERS}*5 )) iterations. Exiting..."
+  end_time=$(date +%s)
+  export WORKERS_UPGRADE_DURATION=$((${end_time} - ${start_time}))
```

Comment:
So even if the worker upgrade fails - we are basically exporting the upgrade duration which is 5 mins * number of workers?

Reply:
Yep, that's right, but it always happen when you reach the timeout. You will send the timeout time, but the upgrade result will be fail, so we will exclude on the dashboards